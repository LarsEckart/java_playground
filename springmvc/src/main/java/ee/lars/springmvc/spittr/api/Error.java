package ee.lars.springmvc.spittr.api;

public class Error {

    private final int code;
    private final String message;

    public Error(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return this.code;
    }

    public String getMessage() {
        return this.message;
    }
}