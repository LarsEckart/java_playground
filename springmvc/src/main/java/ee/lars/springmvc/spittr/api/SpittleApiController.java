package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/spittles")
public class SpittleApiController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleApiController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }

    @RequestMapping(method = RequestMethod.POST, consumes = "application/json")
    public Spittle saveSpittle(@RequestBody Spittle spittle) {
        return spittleRepository.save(spittle);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public ResponseEntity<?> spittleById(@PathVariable long id) {
        Spittle spittle = this.spittleRepository.findOne(id);
        if (spittle == null) {
            Error error = new Error(4, "Spittle [" + id + "] not found");
            return new ResponseEntity<Error>(error, HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<Spittle>(spittle, HttpStatus.OK);
    }
}
