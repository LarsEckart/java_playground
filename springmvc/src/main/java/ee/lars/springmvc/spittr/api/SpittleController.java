package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
@RequestMapping("/spittles")
public class SpittleController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }
}
