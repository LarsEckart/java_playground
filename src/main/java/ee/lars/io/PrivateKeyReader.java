package ee.lars.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.security.KeyFactory;
import java.security.NoSuchAlgorithmException;
import java.security.PrivateKey;
import java.security.spec.EncodedKeySpec;
import java.security.spec.InvalidKeySpecException;
import java.security.spec.PKCS8EncodedKeySpec;

public class PrivateKeyReader {

    /**
     * Reads a private key file which is in PKCS#8 format.
     * For example create a private key via
     * openssl genrsa -out private_key.pem 2048
     * and then convert it via
     * openssl pkcs8 -topk8 -inform PEM -outform DER -in private_key.pem -out private_key.der -nocrypt
     */
    public static PrivateKey get(String filename)
            throws IOException, NoSuchAlgorithmException, InvalidKeySpecException
    {
        byte[] keyBytes = Files.readAllBytes(Paths.get(filename));

        EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyBytes);
        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
        final PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
        return privateKey;
    }
}
