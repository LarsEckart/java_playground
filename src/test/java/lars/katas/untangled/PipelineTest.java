package lars.katas.untangled;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.util.stream.Collectors;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.mockito.Mockito.mock;

class PipelineTest {

  @Test
  void golden_master() {
    CombinationApprovals.verifyAllCombinations(
        this::executeProgram, new Boolean[] {true, false}, TestStatus.values());
  }

  private String executeProgram(boolean buildsSuccessfully, TestStatus noTests) {
    ByteArrayOutputStream baos = new ByteArrayOutputStream();
    System.setOut(new PrintStream(baos, true, java.nio.charset.StandardCharsets.UTF_8));

    Pipeline pipeline = new Pipeline(mock(Config.class), mock(Emailer.class));
    pipeline.run(
        Project.builder()
            .setDeploysSuccessfully(buildsSuccessfully)
            .setTestStatus(noTests)
            .build());

    String response = baos.toString();
    return response.lines().map(l -> l.substring(13)).collect(Collectors.joining("\n"));
  }
}
