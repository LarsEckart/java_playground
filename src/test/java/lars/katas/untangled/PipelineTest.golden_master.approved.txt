[true, NO_TESTS] => [main] INFO  lars.katas.untangled.Pipeline - No tests
[main] INFO  lars.katas.untangled.Pipeline - Deployment successful
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
[true, PASSING_TESTS] => [main] INFO  lars.katas.untangled.Pipeline - Tests passed
[main] INFO  lars.katas.untangled.Pipeline - Deployment successful
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
[true, FAILING_TESTS] => [main] ERROR lars.katas.untangled.Pipeline - Tests failed
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
[false, NO_TESTS] => [main] INFO  lars.katas.untangled.Pipeline - No tests
[main] ERROR lars.katas.untangled.Pipeline - Deployment failed
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
[false, PASSING_TESTS] => [main] INFO  lars.katas.untangled.Pipeline - Tests passed
[main] ERROR lars.katas.untangled.Pipeline - Deployment failed
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
[false, FAILING_TESTS] => [main] ERROR lars.katas.untangled.Pipeline - Tests failed
[main] INFO  lars.katas.untangled.Pipeline - Email disabled 
